src/lceasy.c

```diff
@@ -1009,6 +1097,243 @@ static int lcurl_easy_set_PROGRESSFUNCTION(lua_State *L){
+  lua_pushinteger(L, info->gid       ); lua_setfield(L, -2, "gid"        );
+  lutil_pushint64(L, info->size      ); lua_setfield(L, -2, "size"       );
+  lutil_pushint64(L, info->hardlinks ); lua_setfield(L, -2, "hardlinks"  );
+  lutil_pushint64(L, info->flags     ); lua_setfield(L, -2, "flags"      );
+
+  lua_newtable(L);
+  if(info->strings.time)  { lua_pushstring (L, info->strings.time  ); lua_setfield(L, -2, "time"  ); }
+  if(info->strings.perm)  { lua_pushstring (L, info->strings.perm  ); lua_setfield(L, -2, "perm"  ); }
+  if(info->strings.user)  { lua_pushstring (L, info->strings.user  ); lua_setfield(L, -2, "user"  ); }
+  if(info->strings.group) { lua_pushstring (L, info->strings.group ); lua_setfield(L, -2, "group" ); }
+  if(info->strings.target){ lua_pushstring (L, info->strings.target); lua_setfield(L, -2, "target"); }
+  lua_setfield(L, -2, "strings");
+
+  lua_pushinteger(L, remains);
+
+  if (lua_pcall(L, n+1, LUA_MULTRET, 0)) {
+    assert(lua_gettop(L) >= top);
+    lua_pushlightuserdata(L, (void*)LCURL_ERROR_TAG);
+    lua_insert(L, top + 1);
+    return CURL_CHUNK_BGN_FUNC_FAIL;
+  }
+
+  if(lua_gettop(L) > top){
+    if(lua_isnil(L, top + 1) && (!lua_isnoneornil(L, top + 2))){
+      lua_settop(L, top + 2);
+      lua_remove(L, top + 1);
+      lua_pushlightuserdata(L, (void*)LCURL_ERROR_TAG);
+      lua_insert(L, top + 1);
+      return CURL_CHUNK_BGN_FUNC_FAIL;
+    }
+    ret = lua_toboolean(L, top + 1) ? CURL_CHUNK_BGN_FUNC_OK : CURL_CHUNK_BGN_FUNC_SKIP;
+  }
+
+  lua_settop(L, top);
+  return ret;
+}
+
+static int lcurl_chunk_end_callback(void *arg) {
+  lcurl_easy_t *p = arg;
+  lua_State *L = p->L;
+  int ret = CURL_CHUNK_END_FUNC_OK;
+  int top = lua_gettop(L);
+  int n   = lcurl_util_push_cb(L, &p->chunk_end);
+
+  assert(NULL != p->L);
+
+  if (lua_pcall(L, n-1, LUA_MULTRET, 0)) {
+    assert(lua_gettop(L) >= top);
+    lua_pushlightuserdata(L, (void*)LCURL_ERROR_TAG);
+    lua_insert(L, top + 1);
+    return CURL_CHUNK_END_FUNC_FAIL;
+  }
+
+  if(lua_gettop(L) > top){
+    if(lua_isnil(L, top + 1) && (!lua_isnoneornil(L, top + 2))){
+      lua_settop(L, top + 2);
+      lua_remove(L, top + 1);
+      lua_pushlightuserdata(L, (void*)LCURL_ERROR_TAG);
+      lua_insert(L, top + 1);
+      return CURL_CHUNK_END_FUNC_FAIL;
+    }
+    ret = lua_toboolean(L, top + 1) ? CURL_CHUNK_END_FUNC_OK : CURL_CHUNK_END_FUNC_FAIL;
+  }
+
+  lua_settop(L, top);
+  return ret;
+}
+
+static int lcurl_easy_set_CHUNK_BGN_FUNCTION(lua_State *L){
+  lcurl_easy_t *p = lcurl_geteasy(L);
+  return lcurl_easy_set_callback(L, p, &p->chunk_bgn,
+    CURLOPT_CHUNK_BGN_FUNCTION, CURLOPT_CHUNK_DATA,
+    "chunk_bgn", lcurl_chunk_bgn_callback
+  );
+}
+
+static int lcurl_easy_set_CHUNK_END_FUNCTION(lua_State *L){
+  lcurl_easy_t *p = lcurl_geteasy(L);
+  return lcurl_easy_set_callback(L, p, &p->chunk_end,
+    CURLOPT_CHUNK_END_FUNCTION, CURLOPT_CHUNK_DATA,
+    "chunk_end", lcurl_chunk_bgn_callback
```

Comment:
Really need tests :)
But this is FTP only functions and I do not know how to install ftp on travis without sudo.